doc/sphinx/proof-engine/tactics.rst

```diff
@@ -102,35 +101,37 @@ specified, the default selector is used.
 Bindings
 ~~~~~~~~
 
-Tactics that take a term as an argument may also accept :token:`bindings`
-to instantiate some parameters of the term by name or position.
-The general form of a term with :token:`bindings` is
-:n:`@term__tac with @bindings` where :token:`bindings` can take two different forms:
+Tactics that take a term as an argument may also accept :token:`bindings` to
+specify the values to assign unbound variables in a term.
+Bindings can be given by position or name.  Generally these appear in the form
+:n:`with @bindings`.
 
   .. insertprodn bindings bindings
 
   .. prodn::
-     bindings ::= {+ ( {| @ident | @natural } := @term ) }
-     | {+ @one_term }
+     bindings ::= {+ @one_term }
+     | {+ ( {| @ident | @natural } := @term ) }
 
-+ In the first form, if an :token:`ident` is specified, it must be bound in the
-  type of :n:`@term` and provides the tactic with an instance for the
-  parameter of this name. If a :token:`natural` is specified, it refers to
-  the ``n``-th non-dependent premise of :n:`@term__tac`.
+.. todo PR: "must not be specified" below is correct?
```

Comment:
There is no "must not be specified" left-over, so I don't know what this refers to, but if I assume right, yes, this is correct: when some arguments can be automatically inferred, they must not be specified in the first form.